### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by flysonic10**
DISCUSS BEFORE MERGING

Adds authentication to the Pika publisher.

Currently copies over the auth module from the WS Server in order to resolve the vhost. It'd be better to break this out into a separate module in the future.

This also requires the user to set the vhost to an empty string (as opposed to '/') when using the cloud rabbit server. In this case, the vhost will be resolved by the OAuth server.

Finally, there's no place for service configuration globals, so I'm just using an AUTH_URL in the pika publisher. If we have more instances of this, we should consider break these out into some environment configuration that is different from the user conf files.

### CODE REVIEW

**Comment by flysonic10** [comment]
Hmm.. I should also add the auth changes to the Pika Subscriber. 

Let's not merge until that's done.

File: src/cloudbrain/core/config.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import json
+import os
+
+
+def get_config():
+    core_path = os.path.join(os.getcwd(), 'src/cloudbrain/core')
```
Comment (flysonic10) [inline src/cloudbrain/core/config.py]: This won't work when cloudbrain is a dependency of another project. How can we point to the config file?

**Comment by marionleborgne** [comment]
@flysonic10 I had to modify a couple of conf files. The merge conflicts on the PR are pretty easy to fix, but let me know if you need help.

File: src/cloudbrain/core/config.py (hunk 1 above)
Comment (marionleborgne) [inline src/cloudbrain/core/config.py]: use `pkg_resources.resource_filename`:

```
import os
import simplejson as json
from cloudbrain import core
from pkg_resources import resource_filename



def get_config():
    if os.environ.get("DEV", None):
        config_file = resource_filename(core.__name__, 'config.dev.json')
    else:
        config_file = resource_filename(core.__name__, 'config.json')

    with open(config_file, 'rb') as f:
        json_config = json.load(f)

    auth_url = os.environ.get("AUTH_URL", None) or json_config['authUrl']
    rabbit_host = os.environ.get("RABBIT_HOST", None) \
                  or json_config['rabbitHost']

    config = {
        "authUrl": auth_url,
        "rabbitHost": rabbit_host
    }

    return config

```

**Pull request merged**